File: primes.php
2<?php
$half = intdiv($argv[1], 2);
$sieve = str_repeat('1', $half);
for ($i = 3; $i <= (int)sqrt($argv[1]); $i += 2)
	if ($sieve[intdiv($i, 2)]) for ($j = intdiv($i ** 2, 2); $j < $half; $j += $i)
		$sieve[$j] = '0';
$total = 1;
for ($i = 1; $i < $half; $i++) if ($sieve[$i]) {
	echo ', ', 2 * $i + 1;
	$total++;
}
echo "\nTotal: {$total}\n";
?>

File: primes.r
library(compiler)
invisible(enableJIT(3))
cat(2)
limit = strtoi(commandArgs(trailingOnly=TRUE)[1]) - 1
half = limit / 2
sieve = array(TRUE, half)
total = 1
if (limit > 7)
	for (i in seq(3, as.integer(sqrt(limit)), 2))
		if (sieve[i / 2]) sieve[seq(i^2 / 2, half, i)] = FALSE
if (limit > 2)
	for (i in 1:(half - 1))
		if (sieve[i]) {
			total = total + 1
			cat(",", 2 * i + 1)
		}
cat("\nTotal:", total, "\n")